src/tests/webkit/WebKit_Performance_001.test.js

```diff
@@ -1,35 +1,59 @@
-import { setWebKitUrl, calcAvgFPS, webKitBrowserActions } from '../../commonMethods/commonFunctions'
+import {
+  setWebKitUrl,
+  calcAvgFPS,
+  webKitBrowserActions,
+  pluginActivate,
+  suspendOrResumePlugin,
+  pluginDeactivate,
+} from '../../commonMethods/commonFunctions'
 import { fetchWebKitFPS } from '../../commonMethods/webKitPerformanceCommonFunctions'
 import constants from '../../commonMethods/constants'
 
 let listener
 
 export default {
   context: {
-    minFPS: 40,
-    url: 'http://webkit.org/blog-files/3d-transforms/poster-circle.html',
+    minFPS: 30,
+    url: 'https://webkit.org/blog-files/3d-transforms/poster-circle.html',
   },
   setup() {
     this.$data.write('samples', [])
     return this.$sequence([
-      () => webKitBrowserActions.call(this),
-      () =>
-        (listener = this.$thunder.api.WebKitBrowser.on('urlchange', data => {
-          this.$data.write('currentUrl', data.url)
-        })),
+      () => pluginDeactivate.call(this, 'WebKitBrowser'), //cycle the browser
+      () => pluginDeactivate.call(this, 'UX'), //make sure UX is turned off
+      () => pluginDeactivate.call(this, 'Netflix'), //make sure Netflix is turned off
+      () => pluginDeactivate.call(this, 'Cobalt'), //make sure Cobalt is turned off
+      () => pluginActivate.call(this, 'WebKitBrowser'),
+      () => setWebKitUrl.call(this, 'about:blank'),
+      () => {
+        return this.$thunder.api.call('WebKitBrowser', 'state', 'resumed')
+      },
+      () => {
+        listener = this.$thunder.api.WebKitBrowser.on(
+          'urlchange',
+          data => {
+            this.$log('Got urlchange event: ', data.url)
+            this.$data.write('currentUrl', data.url)
+          },
+          e => {
+            this.$log('Error subscribing to urlchange: ', e)
+          }
+        )
+        return true
+      },
     ])
   },
   teardown() {
     setWebKitUrl.call(this, constants.blankUrl)
-    listener.dispose()
```

Comment:
Don't we need to dispose the listener here?

Reply:
we do! This is a mistake on my side.